msal/src/main/java/com/microsoft/identity/client/internal/controllers/ExceptionAdapter.java

```diff
@@ -40,22 +40,25 @@
 
     public static MsalException exceptionFromAcquireTokenResult(final AcquireTokenResult result) {
         final AuthorizationResult authorizationResult = result.getAuthorizationResult();
-        final AuthorizationErrorResponse authorizationErrorResponse = authorizationResult.getAuthorizationErrorResponse();
-        final TokenResult tokenResult = result.getTokenResult();
-        final TokenErrorResponse tokenErrorResponse;
-
-        if (!authorizationResult.getSuccess()) {
-            //THERE ARE CURRENTLY NO USAGES of INVALID_REQUEST
-            switch (result.getAuthorizationResult().getAuthorizationStatus()) {
-                case FAIL:
-                    return new MsalServiceException(authorizationErrorResponse.getError(), authorizationErrorResponse.getError() + ";"
-                            + authorizationErrorResponse.getErrorDescription(), MsalServiceException.DEFAULT_STATUS_CODE, null);
-                case USER_CANCEL:
-                    return new MsalUserCancelException();
 
+        if (null != authorizationResult) {
+            final AuthorizationErrorResponse authorizationErrorResponse = authorizationResult.getAuthorizationErrorResponse();
+            if (!authorizationResult.getSuccess()) {
+                //THERE ARE CURRENTLY NO USAGES of INVALID_REQUEST
+                switch (result.getAuthorizationResult().getAuthorizationStatus()) {
+                    case FAIL:
+                        return new MsalServiceException(authorizationErrorResponse.getError(), authorizationErrorResponse.getError() + ";"
```

Comment:
Should we do a null check here? 
Can `authorizationErrorResponse` be null in case of `FAIL` ?

Reply:
I looked around (there's a few usages) and nothing stuck-out to me.... if there's any issues, hopefully we can suss them out in our bug bash

![capture](https://user-images.githubusercontent.com/990063/46235892-32ece580-c331-11e8-8b2d-5a1ea71ee805.JPG)

Reply:
Sounds good. Thanks @iambmelt